### PR TITLE
Added test for Cross Entropy derivative

### PR DESCRIPTION
**Pull request by nishnik**
While looking in coveralls, I found that `deriv` for `Crossentropy` is not called even once.

Added proper documentation for loss functions in `other.jl`

If fine, I would extend it to `deriv2`.

### CODE REVIEW

**Comment by Evizero** [comment]
Hi! Yes the crossentropy one is a big of a loner. It is a bit of a hack and doesn't quite fit in the overall design, yet it seems useful to have.

**Comment by nishnik** [comment]
@Evizero is the current implementation of [`test_deriv`](https://github.com/JuliaML/LossFunctions.jl/pull/80/files#diff-e3150612754e2776f821f9ff5457f1b6R143) fine ? 
If yes, I can add a patch for `deriv2`

**Comment by Evizero** [comment]
Looks good to me! does it pass the tests?

**Comment by nishnik** [comment]
It passes locally on my laptop, will have to wait for travis checks.

File: src/supervised/other.jl [hunk 1]
```diff
@@ -26,12 +27,21 @@ isconvex(::PoissonLoss) = true
...
+
+``L(target, output) = - target*ln(output) - (1-target)*ln(1-output)``
+"""
 
 immutable CrossentropyLoss <: SupervisedLoss end
 typealias LogitProbLoss CrossentropyLoss
 
 function value(loss::CrossentropyLoss, target::Number, output::Number)
+    target >= 0 && target <=1 || error("target must be in [0,1]")
+    output >= 0 && output <=1 || error("output must be in [0,1]")
```
Comment (Evizero) [inline src/supervised/other.jl]: I am unsure if we should do such checks. we don't usually do them because of the nature of these functions (they get called very often in some inner loop). So normally we assumer the caller knows what he/she is doing.

I am ok with this approach as well though.

**Comment by nishnik** [comment]
Soft ping @evizero !
Could you please review.

File: src/supervised/other.jl (hunk 1 above)
Comment (tbreloff) [inline src/supervised/other.jl]: I would vote to remove the checks personally.  But I don't feel strongly enough to hold up a merge

Comment (nishnik) [inline src/supervised/other.jl]: @Evizero @tbreloff 
Agreed with your concerns, but I think checks are necessary (espescially for libraries which other libraries are going to use). I would prefer an error rather than a strange behaviour in my code.

There are checks in `PeriodicLoss`, `HuberLoss` and epsilon losses.

Though I am not much experienced but in [SymPy](github.com/sympy/sympy) they have used checks for every possible corner case. Also Julia is awesome in showing errors:
```
ERROR: output must be in [0,1]
in value(::LossFunctions.CrossentropyLoss, ::Int64, ::Int64) at /home/nishnik/.julia/v0.5/LossFunctions/src/supervised/other.jl:44
```
Sorry, I am not much experienced in Julia to strongly pitch this, it is just my thinking. If you want me to remove checks I would do it, but I am a little reluctant to do this.
Otherwise ready to merge from my side.

Comment (Evizero) [inline src/supervised/other.jl]: > There are checks in PeriodicLoss, HuberLoss and epsilon losses.

The difference there is that the check is in the constructor, while here it is in the computational kernel. The later is executed much more often in real use.

Let's keep it for now

**Comment by Evizero** [comment]
Thanks for working on this
